service/consul/environment.go

```diff
@@ -209,13 +208,69 @@ func NewEnvironment(l log.Logger, registrationScheme string, co Options, eo ...s
 		return nil, err
 	}
 
-	return environment{
+	newServiceEnvironment := environment{
 		service.NewEnvironment(
 			append(
 				eo,
 				service.WithRegistrars(r),
 				service.WithInstancers(i),
 				service.WithCloser(closer),
-			)...,
-		), NewClient(consulClient)}, nil
+			)...), NewClient(consulClient)}
+
+	if co.DatacenterWatchInterval > 0 {
+		go WatchInstancers(l, co, newServiceEnvironment)
+	}
+
+	return newServiceEnvironment, nil
+}
+
+func WatchInstancers(l log.Logger, co Options, e Environment) {
```

Comment:
Does this need to be a public function? Presumably those using this package are going to call `NewEnvironment()`, right?

Reply:
The idea behind exposing some of these other functions was that clients might want to reuse pieces of the package.  That wound up not happening.

This is probably a bigger issue.  This area of `webpa-common` really needs some refactoring to remove `zookeeper` support and simplify the watcher code.